### ISSUE TITLE
Cannot use scalar subquery originating from MULTISET in JSON_OBJECT() VALUE clause in Db2

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13681]
Another example of Db2's many failures in correctly implementing SQL/JSON is this query:

```sql
select (
  select coalesce(
    cast((('[' || listagg(
      json_object(key 'v0' value v0 format json),
      ','
    )) || ']') as varchar(32672)),
    json_array()
  )
  from (
    select json_object(
      key 'v0' value 1,
      key 'v1' value (
        select coalesce(
          cast((('[' || listagg(
            json_object(
              key 'v0' value v0,
              key 'v1' value v1,
              key 'v2' value v2,
              key 'v3' value v3
            ),
            ','
          )) || ']') as varchar(32672)),
          json_array()
        )
        from (
          select
            T_COMP_CLIENT_VIRTUAL_1.ID v0,
            T_COMP_CLIENT_VIRTUAL_1.VAL v1,
            1 v2,
            (T_COMP_CLIENT_VIRTUAL_1.VAL * 2) v3
          from T_COMP_CLIENT_VIRTUAL_1
          order by v0
        ) t
      ) format json
    ) v0
    from SYSIBM.DUAL
  ) t
)
from SYSIBM.DUAL;
```

Producing:

> SQL Error [42601]: An unexpected token "format" was found following ") t 
>      )".  Expected tokens may include:  "<interval_qualifier>".. SQLCODE=-104, SQLSTATE=42601,  DRIVER=4.28.11

Workaround:

```sql
select (
  select coalesce(
    cast((('[' || listagg(
      json_object(key 'v0' value v0 format json),
      ','
    )) || ']') as varchar(32672)),
    json_array()
  )
  from (
    select json_object(
      key 'v0' value 1,
      key 'v1' value coalesce(null, ( -- Another coalesce here
        select coalesce(
          cast((('[' || listagg(
            json_object(
              key 'v0' value v0,
              key 'v1' value v1,
              key 'v2' value v2,
              key 'v3' value v3
            ),
            ','
          )) || ']') as varchar(32672)),
          json_array()
        )
        from (
          select
            T_COMP_CLIENT_VIRTUAL_1.ID v0,
            T_COMP_CLIENT_VIRTUAL_1.VAL v1,
            1 v2,
            (T_COMP_CLIENT_VIRTUAL_1.VAL * 2) v3
          from T_COMP_CLIENT_VIRTUAL_1
          order by v0
        ) t
      )) format json
    ) v0
    from SYSIBM.DUAL
  ) t
)
from SYSIBM.DUAL;
```

**Comment by lukaseder** [comment]
It seems this is an instance of #12189 that has not yet been caught...?

**Comment by lukaseder** [comment]
Ah, the fix from #12189 just checks if we passed a `ScalarSubquery` for the `VALUE` clause, but it might as well be something that produces a `ScalarSubquery` through emulation, e.g. `MULTISET`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.7 (https://github.com/jOOQ/jOOQ/issues/13682), and 3.15.11 (https://github.com/jOOQ/jOOQ/issues/13683)
